proposed/2022/WarningsNotAsErrors.md

```diff
@@ -0,0 +1,86 @@
+# Honor WarningsNotAsErrors
+
+- Author Name <https://github.com/nkolev92>
+- Start Date 2022-10-01
+- GitHub Issue <https://github.com/NuGet/Home/issues/5375>
+- GitHub PR <https://github.com/NuGet/NuGet.Client/pull/4846>
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+`WarningsNotAsErrors` is an existing [compiler option](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/errors-warnings) that allows users with treat warnings as errors, to not get errors for every warning raised by their build.
+NuGet supports `TreatWarningsAsErrors`, `WarningsAsErrors` and `NoWarn`. In this proposal we want to add support for `WarningsNotAsErrors`.
+
+## Motivation
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+
+`TreatWarningsAsErrors` is very commonly used feature in the .NET world. NoWarn allows you to suppress certain warning, but that is often too strong. `WarningsNotAsErrors` provides a middle ground, where the users are aware of the warning, but it doesn't fail their build.
+This feature is supported by the compiler, and we want to provide parity for both pack and restore.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+The proposed behavior here matches the compiler behavior. All the scenarios are best represented in a table.
+
+The precedence order for the 3 properties by NuGet can summarized as:
+`NoWarn > WarningsAsErrors > WarningsNotAsErrors`.
+
+For the following table assume `NU1603` is being raised during restore.
+
+| TreatWarningsAsErrors | NoWarn | WarningsAsErrors | WarningsNotAsErrors | Behavior | Explanation |
+|-----------------------|--------|------------------|---------------------|----------|-------------|
+| true | | | | All warnings are treated as errors | |
+| true | | | | All warnings are treated as errors | |
+| true | | NU1603 |  | Succeeds with no warning. | NU1603 is raised, but because of no warn it won't be shown. |

```

Comment:
The NoWarn column is empty, but the explanation says that "because of no warn it won't be shown"